chapters/operatorsandexpressions.tex

```diff
@@ -22,17 +22,29 @@ \section{Expressions}\label{expressions}
 
 \section{Operator Precedence and Associativity}\label{operator-precedence-and-associativity}
 
-Operator precedence determines the order of evaluation of operators in an expression.
-An operator with higher precedence is evaluated before an operator with lower precedence in the same expression.
-For example, `\lstinline!*!' having higher precedence than `\lstinline!+!' means that \lstinline!1 + 2 * 3! becomes evaluated as \lstinline!1 + (2 * 3)!.
-
-Operator associativity are used to determine the order of evaluation when operators have equal precedence.
-Left associativity means that evaluation proceeds from left to right.
-For example, left associativity of binary `\lstinline!-!' means that \lstinline!1 - 2 - 3! becomes evaluated as \lstinline!(1 - 2) - 3!.
-Right associativity means evaluation from right to left, but there are no operators with this associativity in Modelica.
-An operator may also be non-associative, meaning that evaluation order cannot be determined based on associativity.
-For example, non-associativity of `\lstinline!<!' means that \lstinline!1 < 2 < 3! is an invalid expression.
-Note that the operators don't need to be identical for associativity to matter; also \lstinline!1 == 2 < 3! is invalid, and \lstinline!1 - 2 + 3! becomes evaluated as \lstinline!(1 - 2) + 3!.
+Operator precedence determines the implicit subexpression structure of expressions with operators.
+(Explicit subexpression structure can be expressed by wrapping the subexpression in parentheses.)
+An operator with higher precedence ties harder to its operands than an operator with lower precedence.
+For example, `\lstinline!*!' having higher precedence than `\lstinline!+!' means that \lstinline!1 + 2 * 3! is implicitly structured as \lstinline!1 + (2 * 3)!.
+
+Precedence group associativity is used to determine the implicit subexpression structure when operators belong to the same group of equal precedence.
+Left associativity means that subexpressions are formed from left to right.
+For example, left associativity of binary additive operators means that \lstinline!1 - 2 - 3! is implicitly structured as \lstinline!(1 - 2) - 3!.
+Right associativity means forming subexpressions from right to left, but there is no precedence group with this associativity in Modelica.
+A precedence group may also be non-associative, meaning that there is no implicit subexpression structure defined based on associativity.
+For example, non-associativity of relational operators means that \lstinline!1 < 2 < 3! is an invalid expression.
+Note that the operators don't need to be identical for associativity to matter; also \lstinline!1 == 2 < 3! is invalid, and \lstinline!1 - 2 + 3! is implicitly structured as \lstinline!(1 - 2) + 3!.
+
+\begin{nonnormative}
```

Comment:
I think that this entire section about mathematical associativity creates more questions than it answers, and would propose to remove it.

Technically associativity in languages is related to the mathematical associativity, but it would require substantially more explanation.
The case a*b*c for vectors b and c is more complicated: the bi-linearity of the scalar product mean that the two different variants actually give the same result. However if all of a, b, c are vectors the results will differ:
```
{1,0,0}*{1,1,0}*{0,1,0}
 = {0, 1, 0}
{1,0,0}*({1,1,0}*{0,1,0})
 = {1, 0, 0}
```

Reply:
Can we try removing the parts giving rise to questions, while still conveying the main points:
- At the parsing stage – which is where the operator precedence and associativity table matters – the subexpression structure is fixed.
- Once the mapping to mathematical operators is resolved, tools are free to perform mathematically valid rewritings of the the subexpression structure, with a reference to `\cref{evaluation-order}`.

Would you agree that those points are worth mentioning?

Reply:
The first point is important, the second point isn't important. We already have a section on evaluation order and trying to explain that and the mathematics of associativity doesn't add anything for the specification, although it might be important in an annotated reference manual or similarly.